src/borg/testsuite/archiver.py

```diff
@@ -960,6 +960,31 @@ def test_create_pattern_exclude_folder_no_recurse(self):
         self.assert_not_in('input/x/a', output)
         self.assert_in('A input/y/foo_y', output)
 
+    def test_create_pattern_intermediate_folder_permissions(self):
+        """test for correct metadata when patterns exclude a parent folder but include a child"""
+        self.patterns_file_path2 = os.path.join(self.tmpdir, 'patterns2')
+        with open(self.patterns_file_path2, 'wb') as fd:
+            fd.write(b'+ input/x/a\n+ input/x/b\n- input/x*\n')
+
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+
+        self.create_regular_file('x/a/foo_a', size=1024 * 80)
+        self.create_regular_file('x/b/foo_b', size=1024 * 80)
+        with changedir('input'):
+            os.chmod('x/a', 0o750)
+            os.chmod('x/b', 0o700)
+            self.cmd('create', '--patterns-from=' + self.patterns_file_path2,
+                     self.repository_location + '::test', '.')
+
+        # extract the archive and verify that the "intermediate" folder
+        # permissions have been preserved.
+        with changedir('output'):
+            self.cmd('extract', self.repository_location + '::test')
+
+            for fname, mode in [('x/a', 0o750), ('x/b', 0o700)]:
+                st = os.stat(fname)
+                self.assert_equal(st.st_mode & 0o777, mode)
```

Comment:
hmm, wasn't the point of this test to make an assertion about 'x' (which is excluded) rather than 'x/a' and 'x/b', which both are normally included files?